pkgs/servers/keycloak/default.nix

```
@@ -19,11 +19,11 @@ let
 in
 stdenv.mkDerivation rec {
   pname   = "keycloak";
-  version = "16.1.0";
+  version = "17.0.1";
```

```suggestion
  version = "17.0.1-legacy";
```
How do we integrate the non-legacy/regular 17.0.1 version, if we implicitly link keycloak-17.0.1 to the legacy package?

---

Well, I figured we'd just do a hard switch-over - rewrite the module and switch the package at the same time. Unless there are users who use the package without the module, this should work fine. If there are, and they need the legacy package, i guess we need to package both.

---

hmm- I'd prefer a separate version to indicate legacy, and follow the upstream versioning scheme in the version here.
We could add a check to the module to require <17 || 17*-pre, at least until we're able to update the module (and then assert >= 17)..

---

Given https://bugzilla.redhat.com/show_bug.cgi?id=2013577, I can live with either versioning option, however.

---

(I'm not sure if 
```suggestion
  version = "17.0.1.legacy";
```
would be better)